### PR TITLE
Fix Sourcepoint CCPA popup handling

### PR DESCRIPTION
**Pull request by muodov**
https://app.asana.com/0/1201279386299865/1203468538517507/f

### CODE REVIEW

File: lib/cmps/sourcepoint-frame.ts [hunk 1]
```diff
@@ -7,10 +7,11 @@ import AutoConsentCMPBase from "./base";
 export default class SourcePoint extends AutoConsentCMPBase {
   prehideSelectors = ["div[id^='sp_message_container_'],.message-overlay"]
 
-  ccpaMode = false;
+  ccpaNotice = false;
+  ccpaPopup = false;
 
   runContext: RunContext = {
-    main: false,
+    main: true,
```
Comment (sammacbeth) [inline lib/cmps/sourcepoint-frame.ts:14]: Does this popup run in the main frame? Where I can test it?

File: lib/rule-executors.ts [hunk 2]
```diff
@@ -10,9 +10,14 @@ export function doEval(expr: string): Promise<boolean> {
...
-  const elem = document.querySelectorAll<HTMLElement>(selector);
-  enableLogs && console.log("[click]", selector, all, elem);
+export function click(selectorOrElements: string | HTMLElement[], all = false): boolean {
+  let elem: HTMLElement[] = [];
+  if (typeof selectorOrElements === 'string') {
+    elem = Array.from(document.querySelectorAll<HTMLElement>(selectorOrElements));
+  } else {
+    elem = selectorOrElements;
+  }
+  enableLogs && console.log("[click]", selectorOrElements, all, elem);
```
Comment (sammacbeth) [inline lib/rule-executors.ts:20]: I'm not sure that we need to change the `click` API for this use-case. You can just call `t.click()` directly in `sourcepoint-frame.ts`.

File: lib/cmps/sourcepoint-frame.ts (hunk 1 above)
Comment (muodov) [reply lib/cmps/sourcepoint-frame.ts:14]: This is to make the prehide selectors work: https://app.asana.com/0/0/1203411040749543/1203463729387875/f

File: lib/rule-executors.ts (hunk 2 above)
Comment (muodov) [reply lib/rule-executors.ts:20]: Direct `.click()` calls are possible, but they make debugging class-based rules more difficult. They don't produce logs, so it's harder to see where we failed at a glance. And they don't return the success result, which adds boilerplate for non-optional steps.

My idea was to make these utility functions be used both in declarative and class-based rules so both syntaxes provide the same functionality

File: lib/cmps/sourcepoint-frame.ts (hunk 1 above)
Comment (muodov) [reply lib/cmps/sourcepoint-frame.ts:14]: Prehide selectors need to be applied in the top frame. But now I realize that they don't depend on the runContext, so this is indeed not necessary. Thanks for catching this!

**Review by sammacbeth — APPROVED** [review]

**Pull request merged**